kafka-plugins-0.8/src/main/java/co/cask/hydrator/plugin/batch/source/KafkaBatchSource.java

```diff
@@ -85,7 +85,7 @@ public void prepareRun(BatchSourceContext context) throws Exception {
     KafkaPartitionOffsets partitionOffsets = config.getInitialPartitionOffsets();
 
     // If the offset directory is provided, try to load the file
-    if (config.getOffsetDir() != null) {
+    if (!context.isPreviewEnabled() && config.getOffsetDir() != null) {
```

Comment:
should we return before getting `partitionOffsets`?

Reply:
There is still following steps that need to be done to generate kafka request.